src/option.rs

```diff
@@ -19,7 +19,20 @@
 use super::{Error, IterError, Metadata, Miniconf, Peekable};
 
 /// An `Option` that exposes its value through their [`Miniconf`](trait.Miniconf.html) implementation.
-pub struct Option<T>(pub core::option::Option<T>);
+#[derive(
+    Clone,
+    Copy,
+    Default,
+    PartialEq,
+    Eq,
+    Debug,
+    PartialOrd,
+    Ord,
+    Hash,
+    serde::Serialize,
+    serde::Deserialize,
```

Comment:
Similar to above, what if the `T` doesn't impl `Serialize` or `Deserialize`? It's not required to based on our impl, since we defer downward into the `Miniconf` impl

Reply:
It does the right thing.

Reply:
Why derive `Serialize` and `Deserialize` here, but not on `Array`?

Reply:
Can't. Same as default.

Reply:
Ah right, there's the annoying issue of only implementing things through [T; 32] that has held around... https://docs.rs/serde/1.0.147/serde/ser/index.html#implementations-of-serialize-provided-by-serde

I think this could be done via conditional impls if we wanted, but we can deal with this later if it ever is an issue.